### PR TITLE
1.8 client libraries

### PR DESCRIPTION
**Pull request by kelseiv**
Update procedures for writing data using 1.8 client libraries with new `/api/v2/write` endpoint.

### CODE REVIEW

File: content/influxdb/v1.8/guides/write_data.md [hunk 1]
```diff
@@ -24,28 +24,34 @@ curl -i -XPOST http://localhost:8086/query --data-urlencode "q=CREATE DATABASE m
...
+
+For example, to write a single point to the `mydb` database.
+The data consists of the [measurement](/influxdb/v1.8/concepts/glossary/#measurement) `cpu_load_short`, the [tag keys](/influxdb/v1.8/concepts/glossary/#tag-key) `host` and `region` with the [tag values](/influxdb/v1.8/concepts/glossary/#tag-value) `server01` and `us-west`, the [field key](/influxdb/v1.8/concepts/glossary/#field-key) `value` with a [field value](/influxdb/v1.8/concepts/glossary/#field-value) of `0.64`, and the [timestamp](/influxdb/v1.8/concepts/glossary/#timestamp) `1434055562000000000`.
 
 ```bash
 curl -i -XPOST 'http://localhost:8086/write?db=mydb' --data-binary 'cpu_load_short,host=server01,region=us-west value=0.64 1434055562000000000'
+
+//or to write data using new client libraries
+
+curl -i -XPOST 'http://localhost:8086/api/v2/write?db=mydb' --data-binary 'cpu_load_short,host=server01,region=us-west value=0.64 1434055562000000000'
```
Comment (Anaisdg) [inline content/influxdb/v1.8/guides/write_data.md:40]: I think this is a duplicate curl request

File: content/influxdb/v1.8/tools/api_client_libraries.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 title: InfluxDB API client libraries
```
Comment (Anaisdg) [inline content/influxdb/v1.8/tools/api_client_libraries.md:2]: Is this table view intended to have this layout? It seems a little hard to read

File: content/influxdb/v1.7/tools/api_client_libraries.md [hunk 3]
```diff
@@ -11,7 +11,9 @@ menu:
     parent: Tools
 ---
 
-InfluxDB client libraries are developed by the open source community. These client libraries support the InfluxDB API and should be fully compatible with InfluxDB version 1.5. Functionality will vary as there are no standard features that all libraries must implement in order to be listed here.
+InfluxDB client libraries are developed by the open source community. These client libraries support the InfluxDB API and should be fully compatible with InfluxDB 1.5+. Functionality will vary as there are no standard features that all libraries must implement in order to be listed here.
+
+>**Note:** We highly recommend upgrading to InfluxDB 1.8 to use new client libraries compatible with both InfluxDB 1.8 and InfluxDB 2.0. For more information, see [InfluxDB 1.8 API client libraries](/influxdb/v1.8/tools/api-client-libraries).
```
Comment (sanderson) [inline content/influxdb/v1.7/tools/api_client_libraries.md:16]: The link should point to InfluxDB 2.0 Client libraries – https://v2.docs.influxdata.com/v2.0/reference/api/client-libraries/

File: content/influxdb/v1.8/guides/write_data.md (hunk 1 above)
Comment (sanderson) [inline content/influxdb/v1.8/guides/write_data.md:40]: - Client libraries handle all the API requests for you, so the comment here isn't accurate. It should say "... write data using the InfluxDB 2.0 compatible endpoint"
- The v2 write endpoint requires different parameters. @pierwill has already documented these so you can probably just link "`/api/v2/write` endpoint" above to that content. See that PR here: https://github.com/influxdata/docs.influxdata.com/pull/2733/
- I'd also make the curl examples multi-line so they're easier to read:

```bash
curl -i -XPOST 'http://localhost:8086/write?db=mydb' \
  --data-binary 'cpu_load_short,host=server01,region=us-west value=0.64 1434055562000000000'

// or to write data using the InfluxDB 2.0 compatible endpoint

curl -i -XPOST 'http://localhost:8086/api/v2/write?bucket=db/rp&precision=ns' \
  --header 'Authorization: Token username:password' \
  --data-raw 'cpu_load_short,host=server01,region=us-west value=0.64 1434055562000000000'
```

File: content/influxdb/v1.8/tools/api_client_libraries.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
-description: InfluxDB API client libraries includes support for Elixir, Go, Haskell, Java, JavaScript/Node.js, Lisp, MATLAB, .Net, Perl, PHP, Python, R, Ruby, Rust, Scala, Sensu, and the SNMP agent.
```
Comment (sanderson) [inline content/influxdb/v1.8/tools/api_client_libraries.md:3]: Why are we removing all of these.? These 1.x client libraries still exist and can be used to interact with 1.8.

File: content/influxdb/v1.8/tools/api_client_libraries.md [hunk 5]
```diff
@@ -11,131 +11,95 @@ menu:
     parent: Tools
 ---
 
-InfluxDB client libraries are developed by the open source community and InfluxData.
-These client libraries support the InfluxDB API and should be fully compatible with InfluxDB version 1.5+.
-Functionality will vary as there are no standard features that all libraries must implement in order to be listed here.
+InfluxDB client libraries are language-specific packages that integrate with the InfluxDB API and support both **InfluxDB 1.8+** and **InfluxDB 2.0**. The client libraries on this page are available for writing data to InfluxDB using the [`/api/v2/write` endpoint](/influxdb/v1.8/tools/api#api-v2-write-http-endpoint).
```
Comment (sanderson) [inline content/influxdb/v1.8/tools/api_client_libraries.md:14]: All the 1.x client libraries still work with 1.8. Users just have the option of using the newer 2.0 client libraries. 1.x client libraries still use the `/write` API endpoint. Only 2.0 client libraries use the `/api/v2/write` endpoint.

File: content/influxdb/v1.7/tools/api_client_libraries.md (hunk 3 above)
Comment (kelseiv) [reply content/influxdb/v1.7/tools/api_client_libraries.md:16]: Updated link and heading to match v2 "InfluxDB client libraries" and left link pointing to 1.8

File: content/influxdb/v1.8/guides/write_data.md (hunk 1 above)
Comment (kelseiv) [reply content/influxdb/v1.8/guides/write_data.md:40]: Diff between the 2 curl requests is one is writing to the `/api/v2/write`endpoint. I've updated text to clarify this distinction.

File: content/influxdb/v1.8/tools/api_client_libraries.md (hunk 2 above)
Comment (kelseiv) [reply content/influxdb/v1.8/tools/api_client_libraries.md:2]: @Anaisdg, thanks for checking - line 1-12 is part of the front matter, so folks won't actually see most of it...just the title (and option in the Tools menu).

File: content/influxdb/v1.8/tools/api_client_libraries.md (hunk 4 above)
Comment (kelseiv) [reply content/influxdb/v1.8/tools/api_client_libraries.md:3]: Refocused 1.8+ on InfluxDB 2.0-compatible client libraries. Did mention in the 1.7 topic that the "old" client libraries may continue to work, but are not maintained by InfluxData.

File: content/influxdb/v1.8/guides/write_data.md (hunk 1 above)
Comment (sanderson) [reply content/influxdb/v1.8/guides/write_data.md:40]: @kelseiv There are more differences in the new client libraries curl command:

```sh
curl -i -XPOST 'http://localhost:8086/api/v2/write?bucket=db/rp&precision=ns' \
  --header 'Authorization: Token username:password' \
  --data-raw 'cpu_load_short,host=server01,region=us-west value=0.64 1434055562000000000'
```

Comment (kelseiv) [reply content/influxdb/v1.8/guides/write_data.md]: @sanderson, it looks like the same....did I miss something specifically? thanks!

Comment (kelseiv) [reply content/influxdb/v1.8/guides/write_data.md]: here's the commit I updated this one in:

https://github.com/influxdata/docs.influxdata.com/pull/2757/commits/c3cacb211df477009394727ed7bb4790a8763eff

**Comment by awegrzyn** [comment]
Hi,
I'm developer of InfluxDB C++ library (https://github.com/awegrzyn/influxdb-cxx.git). I understand that it does not support v2 protocol yet (I'm almost done with it). I'm really surprised that you removed reference to library without reaching out to me.
Could you clarify what's current policy? Will you provide bindings for all languages? If so, I'll delete the library from GH and not waste my time on migration to v2.

**Pull request merged**